src/API/Reports/Cache.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * REST API Reports Cache.
+ *
+ * Handles report data object caching.
+ *
+ * @package WooCommerce Admin/API
+ */
+
+namespace Automattic\WooCommerce\Admin\API\Reports;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Reports Cache class.
+ *
+ * @package WooCommerce Admin/API
+ */
+class Cache {
+	/**
+	 * Cache version. Used to invalidate all cached values.
+	 */
+	const VERSION_OPTION = 'woocommerce_reports_cache_version';
+
+	/**
+	 * Default version number rollover threshold.
+	 */
+	const DEFAULT_VERSION_LIMIT = 99999;
+
+	/**
+	 * Increase cache version number.
+	 *
+	 * @return bool
+	 */
+	public static function bump_version() {
+		/**
+		 * Upper limit to roll cache version back to zero.
+		 *
+		 * @param int $limit Limit.
+		 */
+		$limit   = apply_filters( 'woocommerce_reports_cache_version_limit', self::DEFAULT_VERSION_LIMIT );
+		$limit   = is_numeric( $limit ) ? (int) $limit : self::DEFAULT_VERSION_LIMIT;
+		$version = ( self::get_version() + 1 ) % $limit;
+
+		return update_option( self::VERSION_OPTION, $version );
+	}
+
+	/**
+	 * Get cache version number.
+	 *
+	 * @return int
+	 */
+	public static function get_version() {
+		$version = (int) get_option( self::VERSION_OPTION, 1 );
+
+		return $version;
+	}
+
+	/**
+	 * Get cached value.
+	 *
+	 * @param string $key Cache key.
+	 * @return mixed
+	 */
+	public static function get( $key ) {
+		$version = self::get_version();
+		$value   = get_transient( $key . ':' . $version );
```

Comment:
This was something I brought up at the GM with @psealock but realised you missed this :)

Using version numbers in transient names causes issues down the line because the transient name is dynamic. It makes cleanup difficult, and leads to lots of stored transient data if those values are not cleaned up regulary.

There is a simple workaround for this. Store the version **within the transient**. 

i.e. the transient data becomes an array like this:

```
[
'version' => X,
'data' => '...original data here...',
]
```

Instead of then reading a transient named `$key . ':' . $version`, you read a transient named `$key` and then check the version that was stored. If it differs, regenerate the data (its invalid). If the version matches, we can return the data (its valid).